src/main/java/com/jcabi/log/Logger.java

```diff
@@ -429,6 +429,32 @@ public void write(final int data) throws IOException {
     }
 
     /**
+     * Log messages constructed from Suppliers (Java 8 users).
```

Comment:
@amihaiemil why we are limiting it to Java 8 users? In Java 7 you can just provide interface implementation I think